contracts/staking/stakeManager/StakeManager.sol

```diff
@@ -389,6 +326,21 @@ contract StakeManager is IStakeManager, StakeManagerStorage {
         CHECKPOINT_REWARD = newReward;
     }
 
+    // Change delegation contract for a validator
+    // @note: Users must exit before this update or all funds may get lost
+    function updateContractAddress(uint256 validatorId, address newContractAddress) public onlyOwner {
+        require(
```

Comment:
what if previous contract will require shares balance to be 0 ?

Reply:
@imyourm8 we shouldn't add that condition. So let's say in the worst condition, governance contracts can drain the balance and deploy contract and upgrade contract address if delegators can't withdraw funds.